integrations/ultralytics/deepsparse/benchmark.py

```diff
@@ -0,0 +1,430 @@
+                        object loaded or loaded into 'model' in a state dict
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -e {deepsparse,onnxruntime,torch}, --engine {deepsparse,onnxruntime,torch}
+                        Inference engine backend to run benchmark on. Choices
+                        are 'deepsparse', 'onnxruntime', and 'torch'. Default
+                        is 'deepsparse'
+  --data-path DATA_PATH
+                        Optional filepath to image examples to run the
+                        benchmark on. Can be path to directory, single image
+                        jpg file, or a glob path. All files should be in jpg
+                        format. If not provided, sample COCO images will be
+                        downloaded from the SparseZoo
+  --image-shape IMAGE_SHAPE [IMAGE_SHAPE ...]
+                        Image shape to benchmark with, must be two integers.
+                        Default is 640 640
+  -b BATCH_SIZE, --batch-size BATCH_SIZE
+                        The batch size to run the benchmark for
+  -c NUM_CORES, --num-cores NUM_CORES
+                        The number of physical cores to run the benchmark on,
+                        defaults to all physical cores available on the system
+  -i NUM_ITERATIONS, --num-iterations NUM_ITERATIONS
+                        The number of iterations the benchmark will be run for
+  -w NUM_WARMUP_ITERATIONS, --num-warmup-iterations NUM_WARMUP_ITERATIONS
+                        The number of warmup iterations that will be executed
+                        before the actual benchmarking
+  -q, --quantized-inputs
+                        Set flag to execute benchmark with int8 inputs instead
+                        of float32
+  --fp16                Set flag to execute torch benchmark in half precision
+                        (fp16)
+  --device DEVICE       Device to benchmark the model with. Default is cpu.
+                        Non cpu benchmarking only supported for torch
+                        benchmarking. Default is 'cpu'
+
+##########
+Example command for running a benchmark on a pruned quantized YOLOv3:
+python benchmark.py \
+    zoo:cv/detection/yolo_v3-spp/pytorch/ultralytics/coco/pruned_quant-aggressive_90 \
+    --batch-size 32 \
+
+##########
+Example for benchmarking on a local YOLOv3 PyTorch model on GPU with half precision:
+python benchmark.py \
+    /PATH/TO/yolov3-spp.pt \
+    --engine torch \
+    --batch-size 32 \
+    --device cuda \
+    --half-precision
+
+##########
+Example for benchmarking on a local YOLOv3 ONNX with onnxruntime:
+python benchmark.py \
+    /PATH/TO/yolov3-spp.onnx \
+    --engine onnxruntime \
+    --batch-size 32 \
+"""
+
+
+import argparse
+import glob
+import os
+import time
+from pathlib import Path
+from typing import Any, Iterable, List, Optional, Tuple, Union
+
+import numpy
+import onnx
+import onnxruntime
+import torch
+from tqdm.auto import tqdm
+
+import cv2
```

Comment:
Need to add cv2 to known third party for setup config and then run make style again